src/cryptsetup/cryptsetup-generator.c

```diff
@@ -27,8 +27,7 @@
 
 typedef struct crypto_device {
         char *uuid;
-        char *keyfile;
-        char *keydev;
+        char *keyspec;
```

Comment:
please keep this separate: parse when parsing, don't parse when applying. this means: when we parse crypttab we should split all these fields apart and then have them split apart already when we apply them. Don't mix applying and parsing!